src/services/game-logic/GameEngine.ts

```diff
@@ -10,7 +12,16 @@ class GameEngine {
 
     getLegalMoves = (square: Square): Square[] => {
         const piece = this.board.getPiece(square);
-        return piece?.getPossibleMoves() ?? [];
+        let pieceMoves = piece?.getPossibleMoves() ?? [];
+        return pieceMoves.filter((square) => this.isOccupiedBySameColorPiece(square, piece));
```

Comment:
to chyba powinno być zaprzeczone? w sensie filter zostawia elementy które spełniają warunek, czyli patrząc na to co jest napisane zostawiamy możliwe ruchy tam gdzie stoją pionki tego samego koloru

Reply:
może to kwestia niefortunnej nazwy. Bo ta funkcja is Occupied... zwraca true, jeśli można wykonać ruch, a można go wykonać, jeśli bierka na dostępnym polu jest innego koloru.

Reply:
tak, nazwa jest do poprawy, odwrotność tam powinna być, czyli isOccupiedByDifferentColorPiece czy enemyColor.
+ ta metoda "kłamie" bo robi dodatkowo sprawdzenie czy nie jest pionkiem :) sugeruję rozbicie na 2 metody, np isPawn(piece) i użycie po prostu 2x filter z !isPawn i isOccupied...

Reply:
@Nebrakada dzięki, zmieniłam, jeśli ok, to poproszę o approve ;)